code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm

```diff
@@ -280,60 +338,276 @@
+					if (CULT_ACT_I)
+						to_chat(user, "<span class='game say'><span class='danger'>Nar-Sie</span> murmurs, <span class='sinister'>The conversion rune is <span class='danger'>Join Blood Self</span>, but you now have many new runes at your disposal to help you in your task, therefore I recommend you first summon an Arcane Tome to easily scribe them. The rune that conjures a tome is <span class='danger'>See Blood Hell</span>.</span></span>")
+					if (CULT_ACT_II)
+						to_chat(user, "<span class='game say'><span class='danger'>Nar-Sie</span> murmurs, <span class='sinister'>To perform the sacrifice, you'll have to forge a cult blade first. It doesn't matter if the target is alive of not, lay their body down on the altar and plant the blade on their stomach. Next, touch the altar to perform the next step of the ritual. The more of you, the quicker it will be done.</span></span>")
+					if (CULT_ACT_III)
+						to_chat(user, "<span class='game say'><span class='danger'>Nar-Sie</span> murmurs, <span class='sinister'>The crew is now aware of our presence, prepare to draw blood. Your priority is to spill as much blood as you can all over the station, bloody trails left by foot steps count toward this goal. How you obtain the blood, I leave to your ambition, but remember that if the crew destroys every blood stones, you will be doomed.</span></span>")
+					if (CULT_ACT_IV)
+						to_chat(user, "<span class='game say'><span class='danger'>Nar-Sie</span> murmurs, <span class='sinister'>One of the blood stones has become my anchor in this plane, you can touch any other stone to locate it. Touch the anchor to perform the Tear Reality ritual before the crew breaks it.</span></span>")
+					if (CULT_EPILOGUE)
+						to_chat(user, "<span class='game say'><span class='danger'>Nar-Sie</span> murmurs, <span class='sinister'>Remarkable work, [user.real_name], I greatly enjoyed observing this game. Your work is over now, but I may have more in store for you in the future. In the meanwhile, bask in your victory.</span></span>")
+				/* TODO: I'll finish that up someday
+				var/dat = {"<body style="color:#FFFFFF" bgcolor="#110000"><ul>"}
+				dat += {"</ul></body>"}
+				user << browse("<TITLE>Nar-Sie's Tips</TITLE>[dat]", "window=narsietips;size=500x300")
+				onclose(user, "narsietips")
+				*/
+			if ("Conjure Soul Gem")
+				altar_task = ALTARTASK_GEM
+				update_icon()
+				overlays += "altar-soulstone1"
+				spawn (gem_delay/3)
+					update_icon()
+					overlays += "altar-soulstone2"
+					sleep (gem_delay/3)
+					update_icon()
+					overlays += "altar-soulstone3"
+					sleep (gem_delay/3)
+					altar_task = ALTARTASK_NONE
+					update_icon()
+					var/obj/item/device/soulstone/gem/gem = new (loc)
+					gem.pixel_y = 4
+
+/obj/structure/cult/altar/noncultist_act(var/mob/user)//Non-cultists can still remove blades planted on altars.
+	if(iscultist(user))
```

Comment:
:thinking: this is noncultist_act, which is specifically called if `iscultist` failed